packages/flutter_tools/templates/plugin_shared/pubspec.yaml.tmpl

```diff
@@ -5,7 +5,12 @@ homepage:
 
 environment:
   sdk: {{dartSdkVersionBounds}}
+{{#withPlatformChannelPluginHook}}
   flutter: ">=2.5.0"
+{{/withPlatformChannelPluginHook}}
+{{#withFfiPluginHook}}
+  flutter: ">={{frameworkVersion}}"
```

Comment:
I'm not sure about this inconsistency; this will mean that (forever, not just right now) by default someone on master creating a platform channel plugin will work for ~everyone, but someone making an FFI plugin will get a plugin that works for almost nobody. I think we should consider making this hard-coded, or maybe making this "minFrameworkVersion" which is populated in code as the earlier of 2.11.0 and the framework version (if we want to exactly handle the version when it lands), with a TODO to switch it to hard-coded later.